### PR TITLE
Add variable auto-completion feature when cursor inside jinja inline brackets (#552)Co-authored-by: pre-commit-ci[bot] <66853113+pre-commit-ci[bot]@users.noreply.github.com>

### PR DESCRIPTION
**Pull request by priyamsahoo**
This PR incorporates the feature to provide variable completions ***(based on scopes: task, block, and play level)*** defined within the playbook when the cursor is inside jinja inline brackets.
Two sets of work and tests are written for this:
1. A function to check if the cursor is within jinja brackets and in the correct yaml format.
2. A function to provide auto-completion when the above function returns true.

Currently, auto-completion works:
1. For variables defined inside the current playbook with valid scopes.
2. For variables defined using keywords: 
	- *vars* at task, block, and play level
	- *vars_prompt*, *vars_files* at play level
3. For variables defined in another file and the file included at the play level, using *vars_files*
4. For cursor locations only inside jinja inline brackets

### CODE REVIEW

File: .mocharc.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 require:
   - ts-node/register
   - test/rootMochaHooks.ts
-reporter: tap # more flat, also the only one that mentions totals for found, skipped and passed
+# reporter: tap # more flat, also the only one that mentions totals for found, skipped and passed
```
Comment (ganeshrn) [inline .mocharc.yml:4]: Is this comment required?

**Review by ganeshrn — APPROVED** [review]
Apart from minor comment the PR looks good to me.

File: .mocharc.yml (hunk 1 above)
Comment (priyamsahoo) [reply .mocharc.yml:4]: I would keep that commented. Because we might have to change to this setting if the log outputs do not look good.

**Comment by priyamsahoo** [comment]
The EE tests for ansible-lint validation are failing because of ansible-lint update in the creator-ee. Those failing tests are unrelated and in no way were caused by the current changes in the PR.

The fixing of those tests can be done in a separate PR.

**Pull request merged**